Makefile

```diff
@@ -1,7 +1,199 @@
-all:
-	$(MAKE) -C src $@
+.POSIX:
 
-%: FORCE
-	$(MAKE) -C src $@
+CXX = c++
 
-.PHONY: all FORCE
+debug = no
+static = no
+gzip_man = yes
+# to get format compatible with GitHub archive use "gzip -S .gz" here
+compress_bin = bzip
+
+compress-suffix-bzip = bz2
+compress-suffix-zstd = zst
+
+CPPFLAGS-debug-yes = -DKAK_DEBUG
+CXXFLAGS-debug-yes = -O0 -g
+suffix-debug-yes = .debug
+
+CXXFLAGS-debug-no = -O3
+suffix-debug-no = .opt
+
+CXXFLAGS-sanitize-address = -fsanitize=address
+LDFLAGS-sanitize-address = -lasan
+suffix-sanitize-address = .san_a
+
+CXXFLAGS-sanitize-undefined = -fsanitize=undefined
+
+LDFLAGS-sanitize-undefined = -lubsan
+suffix-sanitize-undefined = .san_u
+
+LDFLAGS-static-yes = -static -pthread
+
+version != cat .version 2>/dev/null || git describe --tags HEAD 2>/dev/null || echo unknown
+
+sources != find src -type f -name '*.cc' | sed -e '/\.version\.cc/d'
+deps != touch src/main.d && find src -type f -name '*.d'
+objects = $(sources:.cc=.o)
+
+PREFIX = /usr/local
+DESTDIR = # root dir
+
+bindir = $(DESTDIR)$(PREFIX)/bin
+libexecdir = $(DESTDIR)$(PREFIX)/libexec/kak
+sharedir = $(DESTDIR)$(PREFIX)/share/kak
+docdir = $(DESTDIR)$(PREFIX)/share/doc/kak
+mandir = $(DESTDIR)$(PREFIX)/share/man/man1
+
+os != uname
+
+CPPFLAGS-os-Darwin = -I/opt/local/include
+LDFLAGS-os-Darwin = -L/opt/local/lib
+
+CPPFLAGS-os-FreeBSD = -I/usr/local/include
+LDFLAGS-os-FreeBSD = -L/usr/local/lib
+
+LIBS-os-Haiku = -lnetwork -lbe
+
+CPPFLAGS-os-OpenBSD = -DKAK_BIN_PATH="$(bindir)/kak" -I/usr/local/include
+LDFLAGS-os-OpenBSD = -L/usr/local/lib
+mandir-os-OpenBSD = $(DESTDIR)$(PREFIX)/man/man1
+
+LDFLAGS-os-SunOS = -lsocket -rdynamic
+
+is_windows != uname | grep -o '_NT'
+CPPFLAGS-is_windows-_NT = -D_XOPEN_SOURCE=700
+LIBS-is_windows-_NT = -ldbghelp
```

Comment:
Is it possible to merge this into the `...-os-...` support? possibly by sed-ing uname when setting os ?

Reply:
This one ?